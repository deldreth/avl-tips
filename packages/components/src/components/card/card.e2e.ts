import { newE2EPage } from "@stencil/core/testing";

describe("avl-tips-card", () => {
  it("renders", async () => {
    const page = await newE2EPage();

    await page.setContent("<avl-tips-card></avl-tips-card>");
    const element = await page.find("avl-tips-card");
    expect(element).toHaveClass("hydrated");
  });
});
