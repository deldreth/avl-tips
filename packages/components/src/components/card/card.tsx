import { Component, Prop, h } from "@stencil/core";

@Component({
  tag: "avl-tips-card",
  styleUrl: "card.sass",
  shadow: true
})
export class Card {
  @Prop() name: string;
  @Prop() venmo: string;
  @Prop() cash: string;
  @Prop() paypal: string;
  @Prop() employer: string;

  render() {
    return (
      <div class="card employee">
        <div class="card-content">
          Support
          <p class="title">{this.name}</p>
          <p class="subtitle">
            <b>Employer:</b> {this.employer}
          </p>
          <div class="columns is-multiline">
            {this.venmo && (
              <p class="column">
                <a
                  class="button is-info is-medium is-flex"
                  target="_blank"
                  rel="noreferrer noopener"
                  href={`https://venmo.com/${this.venmo.substr(1)}`}
                >
                  Venmo {this.venmo}
                </a>
              </p>
            )}
            {this.cash && (
              <p class="column">
                <a
                  class="button is-success is-medium is-flex"
                  href={`https://cash.app/${this.cash}`}
                  target="_blank"
                  rel="noreferrer noopener"
                >
                  Cash {this.cash}
                </a>
              </p>
            )}
            {this.paypal && (
              <p class="column">
                <a
                  class="button is-primary is-medium is-flex"
                  href={`https://paypal.me/${this.paypal}`}
                  target="_blank"
                  rel="noreferrer noopener"
                >
                  PayPal.Me {this.paypal}
                </a>
              </p>
            )}
          </div>
        </div>
      </div>
    );
  }
}
