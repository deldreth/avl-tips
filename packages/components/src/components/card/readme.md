# my-component



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type     | Default     |
| ---------- | ---------- | ----------- | -------- | ----------- |
| `cash`     | `cash`     |             | `string` | `undefined` |
| `employer` | `employer` |             | `string` | `undefined` |
| `name`     | `name`     |             | `string` | `undefined` |
| `paypal`   | `paypal`   |             | `string` | `undefined` |
| `venmo`    | `venmo`    |             | `string` | `undefined` |


## Dependencies

### Used by

 - [avl-tips](../tips)

### Graph
```mermaid
graph TD;
  avl-tips --> avl-tips-card
  style avl-tips-card fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
