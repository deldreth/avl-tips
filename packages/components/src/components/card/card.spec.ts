import { newSpecPage } from "@stencil/core/testing";
import { Card } from "./card";

describe("avl-tips-card", () => {
  describe("with venmo", () => {
    it("matches snapshot", async () => {
      const page = await newSpecPage({
        components: [Card],
        html:
          "<avl-tips-card name='Test Name' venmo='Venmo' employer='Employer'></avl-tips-card>"
      });

      expect(page.root).toMatchSnapshot();
    });
  });

  describe("with cash", () => {
    it("matches snapshot", async () => {
      const page = await newSpecPage({
        components: [Card],
        html:
          "<avl-tips-card name='Test Name' cash='Cash' employer='Employer'></avl-tips-card>"
      });

      expect(page.root).toMatchSnapshot();
    });
  });

  describe("with paypal", () => {
    it("matches snapshot", async () => {
      const page = await newSpecPage({
        components: [Card],
        html:
          "<avl-tips-card name='Test Name' paypal='Paypal' employer='Employer'></avl-tips-card>"
      });

      expect(page.root).toMatchSnapshot();
    });
  });

  describe("with all payment options", () => {
    it("matches snapshot", async () => {
      const page = await newSpecPage({
        components: [Card],
        html:
          "<avl-tips-card name='Test Name' venmo='Venmo' cash='Cash' paypal='Paypal' employer='Employer'></avl-tips-card>"
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
