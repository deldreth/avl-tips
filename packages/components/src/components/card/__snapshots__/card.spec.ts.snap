// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avl-tips-card with all payment options matches snapshot 1`] = `
<avl-tips-card cash="Cash" employer="Employer" name="Test Name" paypal="Paypal" venmo="Venmo">
  <mock:shadow-root>
    <div class="card employee">
      <div class="card-content">
        Support
        <p class="title">
          Test Name
        </p>
        <p class="subtitle">
          <b>
            Employer:
          </b>
          Employer
        </p>
        <div class="columns is-multiline">
          <p class="column">
            <a class="button is-flex is-info is-medium" href="https://venmo.com/enmo" rel="noreferrer noopener" target="_blank">
              Venmo Venmo
            </a>
          </p>
          <p class="column">
            <a class="button is-flex is-medium is-success" href="https://cash.app/Cash" rel="noreferrer noopener" target="_blank">
              Cash Cash
            </a>
          </p>
          <p class="column">
            <a class="button is-flex is-medium is-primary" href="https://paypal.me/Paypal" rel="noreferrer noopener" target="_blank">
              PayPal.Me Paypal
            </a>
          </p>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</avl-tips-card>
`;

exports[`avl-tips-card with cash matches snapshot 1`] = `
<avl-tips-card cash="Cash" employer="Employer" name="Test Name">
  <mock:shadow-root>
    <div class="card employee">
      <div class="card-content">
        Support
        <p class="title">
          Test Name
        </p>
        <p class="subtitle">
          <b>
            Employer:
          </b>
          Employer
        </p>
        <div class="columns is-multiline">
          <p class="column">
            <a class="button is-flex is-medium is-success" href="https://cash.app/Cash" rel="noreferrer noopener" target="_blank">
              Cash Cash
            </a>
          </p>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</avl-tips-card>
`;

exports[`avl-tips-card with paypal matches snapshot 1`] = `
<avl-tips-card employer="Employer" name="Test Name" paypal="Paypal">
  <mock:shadow-root>
    <div class="card employee">
      <div class="card-content">
        Support
        <p class="title">
          Test Name
        </p>
        <p class="subtitle">
          <b>
            Employer:
          </b>
          Employer
        </p>
        <div class="columns is-multiline">
          <p class="column">
            <a class="button is-flex is-medium is-primary" href="https://paypal.me/Paypal" rel="noreferrer noopener" target="_blank">
              PayPal.Me Paypal
            </a>
          </p>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</avl-tips-card>
`;

exports[`avl-tips-card with venmo matches snapshot 1`] = `
<avl-tips-card employer="Employer" name="Test Name" venmo="Venmo">
  <mock:shadow-root>
    <div class="card employee">
      <div class="card-content">
        Support
        <p class="title">
          Test Name
        </p>
        <p class="subtitle">
          <b>
            Employer:
          </b>
          Employer
        </p>
        <div class="columns is-multiline">
          <p class="column">
            <a class="button is-flex is-info is-medium" href="https://venmo.com/enmo" rel="noreferrer noopener" target="_blank">
              Venmo Venmo
            </a>
          </p>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</avl-tips-card>
`;
